sdk/samples/iot/paho_iot_pnp_component_sample.c

```diff
@@ -0,0 +1,626 @@
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Failed to initialize pnp client");
+
+  // Get the MQTT client id used for the MQTT connection.
+  char mqtt_client_id_buffer[CLIENT_ID_BUFFER_LENGTH];
+
+  rc = az_iot_hub_client_get_client_id(
+      &hub_client, mqtt_client_id_buffer, sizeof(mqtt_client_id_buffer), NULL);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Failed to get MQTT client id");
+
+  // Create the Paho MQTT client.
+  rc = MQTTClient_create(
+      &mqtt_client, mqtt_endpoint_buffer, mqtt_client_id_buffer, MQTTCLIENT_PERSISTENCE_NONE, NULL);
+  if (rc != MQTTCLIENT_SUCCESS)
+  {
+    IOT_SAMPLE_LOG_ERROR("Failed to create MQTT client: MQTTClient return code %d.", rc);
+    exit(rc);
+  }
+}
+
+// connect_mqtt_client_to_iot_hub sets up basic security and MQTT topics and then
+// initiates an MQTT connection to Azure IoT Hub.
+static void connect_mqtt_client_to_iot_hub(void)
+{
+  // Get the MQTT client username.
+  az_result rc = az_iot_hub_client_get_user_name(
+      &hub_client, mqtt_client_username_buffer, sizeof(mqtt_client_username_buffer), NULL);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Failed to get MQTT client username");
+
+  // Set MQTT connection options.
+  MQTTClient_connectOptions mqtt_connect_options = MQTTClient_connectOptions_initializer;
+  mqtt_connect_options.username = mqtt_client_username_buffer;
+  mqtt_connect_options.password = NULL; // This sample uses x509 authentication.
+  mqtt_connect_options.cleansession = false; // Set to false so can receive any pending messages.
+  mqtt_connect_options.keepAliveInterval = AZ_IOT_DEFAULT_MQTT_CONNECT_KEEPALIVE_SECONDS;
+
+  MQTTClient_SSLOptions mqtt_ssl_options = MQTTClient_SSLOptions_initializer;
+  mqtt_ssl_options.verify = 1;
+  mqtt_ssl_options.enableServerCertAuth = 1;
+  mqtt_ssl_options.keyStore = (char*)az_span_ptr(env_vars.x509_cert_pem_file_path);
+  if (az_span_size(env_vars.x509_trust_pem_file_path) != 0) // Is only set if required by OS.
+  {
+    mqtt_ssl_options.trustStore = (char*)az_span_ptr(env_vars.x509_trust_pem_file_path);
+  }
+  mqtt_connect_options.ssl = &mqtt_ssl_options;
+
+  // Connect MQTT client to the Azure IoT Hub.  This will block until the connection
+  // is established or fails.
+  rc = MQTTClient_connect(mqtt_client, &mqtt_connect_options);
+  if (rc != MQTTCLIENT_SUCCESS)
+  {
+    IOT_SAMPLE_LOG_ERROR(
+        "Failed to connect: MQTTClient return code %d.\n"
+        "If on Windows, confirm the AZ_IOT_DEVICE_X509_TRUST_PEM_FILE_PATH environment variable is "
+        "set correctly.",
+        rc);
+    exit(rc);
+  }
+}
+
+// subscribe_mqtt_client_to_iot_hub_topics subscribes to well-known MQTT topics that Azure IoT Hub
+// uses to signal incoming commands to the device and notify device of properties.
+static void subscribe_mqtt_client_to_iot_hub_topics(void)
+{
+  // Subscribe to incoming commands.
+  int rc = MQTTClient_subscribe(
+      mqtt_client, AZ_IOT_HUB_CLIENT_COMMANDS_SUBSCRIBE_TOPIC, IOT_SAMPLE_MQTT_SUBSCRIBE_QOS);
```

Comment:
>IOT_SAMPLE_MQTT_SUBSCRIBE_QOS [](http://example.com/codeflow?start=63&length=29)

Should we have QoS1 or QoS0 in name of this ?

Reply:
Fixing.

Some additional notes -- I got this variable from `iot_sample_common.h`, but... everywhere else in our samples for `MQTTClient_subscribe` we're pasing in 1 directly.  I'm changing this code to follow the convention generally used in samples.